src/chunk.c

```diff
@@ -4359,3 +4366,113 @@ ts_chunk_scan_iterator_set_chunk_id(ScanIterator *it, int32 chunk_id)
 								   F_INT4EQ,
 								   Int32GetDatum(chunk_id));
 }
+
+#include "hypercube.h"
+static Hypercube *
+fill_hypercube_for_foreign_table_chunk(Hyperspace *hs)
+{
+	Hypercube *cube = ts_hypercube_alloc(hs->num_dimensions);
+	Point *p = ts_point_create(hs->num_dimensions);
+	Assert(hs->num_dimensions == 1); // does not work with partitioned range
+	for (int i = 0; i < hs->num_dimensions; i++)
+	{
+		const Dimension *dim = &hs->dimensions[i];
+		Assert(dim->type == DIMENSION_TYPE_OPEN);
+		Oid dimtype = ts_dimension_get_partition_type(dim);
+		// Datum val = ts_time_datum_get_nobegin_or_min(dimtype);
+		Datum val = ts_time_get_min(dimtype);
+		p->coordinates[p->num_coords++] = ts_time_value_to_internal(val, dimtype);
+		cube->slices[i] = ts_dimension_calculate_default_slice(dim, p->coordinates[i]);
```

Comment:
So this chunk gets the "earliest" time slice, right? This means you can't create two such chunks, because chunks don't intersect? I'd expect that `dimension_slice` for the chunk to be a sensible range corresponding to the chunk data.

Reply:
This is a dummy dimension slice - it is not based on the actual data. The foreign table manages its own dimension/slice data. But we cannot a chunk to a hypertable without creating a dimension for it.

Reply:
So the normal chunk exclusion is not going to work for hypertables with these foreign chunks, and it's performed separately by the OSM extension? Would be interesting to read a design doc for this, if there's one.